activerecord/test/cases/schema_loading_test.rb

```diff
@@ -1,52 +1,41 @@
 require "cases/helper"
 
-module SchemaLoadCounter
-  extend ActiveSupport::Concern
-
-  module ClassMethods
-    attr_accessor :load_schema_calls
+class SchemaLoadingTest < ActiveRecord::TestCase
+  self.use_transactional_tests = false
 
-    def load_schema!
-      self.load_schema_calls ||= 0
-      self.load_schema_calls +=1
-      super
-    end
+  setup do
+    ActiveRecord::SchemaMigration.drop_table
+    ActiveRecord::SchemaMigration.create_table
   end
-end
 
-class SchemaLoadingTest < ActiveRecord::TestCase
-  def test_basic_model_is_loaded_once
-    klass = define_model
-    klass.new
-    assert_equal 1, klass.load_schema_calls
+  teardown do
+    ActiveRecord::SchemaMigration.drop_table
   end
 
-  def test_model_with_custom_lock_is_loaded_once
-    klass = define_model do |c|
-      c.table_name = :lock_without_defaults_cust
-      c.locking_column = :custom_lock_version
-    end
-    klass.new
-    assert_equal 1, klass.load_schema_calls
-  end
+  def test_assume_migrated_upto_version_gets_all_versions
+    migrations_path = MIGRATIONS_ROOT + "/valid_with_subdirectories"
 
-  def test_model_with_changed_custom_lock_is_loaded_twice
-    klass = define_model do |c|
-      c.table_name = :lock_without_defaults_cust
-    end
-    klass.new
-    klass.locking_column = :custom_lock_version
-    klass.new
-    assert_equal 2, klass.load_schema_calls
+    ActiveRecord::Base.connection.assume_migrated_upto_version(3, migrations_path)
+    assert_equal 3, ActiveRecord::Migrator.current_version
+
+    sm_table = ActiveRecord::Migrator.schema_migrations_table_name
+    migrated = ActiveRecord::Base.connection.select_values("SELECT version FROM #{sm_table}").map(&:to_i)
+    assert_equal [1,2,3], migrated.sort
   end
 
-  private
+  if current_adapter?(:SQLite3Adapter)
+    %w{3.7.8 3.7.11 3.7.12}.each do |version_string|
```

Comment:
That test is stubbing the connection version too. https://github.com/rails/rails/blob/master/activerecord/test/cases/schema_dumper_test.rb#L35-L36